File: articles/linear-algebra-for-systems-modeling/articles/network-adjacency-matrices/calculators/python/network_adjacency_calculator.py
from __future__ import annotations
import csv, json
from pathlib import Path

def main():
    output_dir = Path("outputs"); output_dir.mkdir(parents=True, exist_ok=True)
    result = {"calculator":"network_adjacency_calculator","network_name":"synthetic_infrastructure_dependency_network","node_count":5,"edge_count":20,"directed":True,"weighted":True,"density":0.8,"max_out_weight":2.15,"max_in_weight":1.95,"row_normalized":True,"warning":"Adjacency metrics depend on node boundaries, edge definitions, direction conventions, weight meaning, and data provenance."}
    (output_dir/"network_adjacency_calculator.json").write_text(json.dumps(result, indent=2, sort_keys=True), encoding="utf-8")
    with (output_dir/"network_adjacency_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys())); writer.writeheader(); writer.writerow(result)
    print(json.dumps(result, indent=2, sort_keys=True))
if __name__ == "__main__": main()

File: articles/linear-algebra-for-systems-modeling/articles/network-adjacency-matrices/calculators/r/network_adjacency_calculator.R
result <- data.frame(calculator="network_adjacency_calculator", network_name="synthetic_infrastructure_dependency_network", node_count=5, edge_count=20, directed=TRUE, weighted=TRUE, density=0.8, max_out_weight=2.15, max_in_weight=1.95, row_normalized=TRUE, warning="Adjacency metrics depend on node boundaries, edge definitions, direction conventions, weight meaning, and data provenance.")
dir.create("outputs", recursive=TRUE, showWarnings=FALSE)
write.csv(result, "outputs/r_network_adjacency_calculator.csv", row.names=FALSE)
print(result)
